app/grandchallenge/archives/tasks.py

```diff
@@ -29,3 +33,10 @@ def add_images_to_archive(*, upload_session_pk, archive_pk):
             continue
         item = ArchiveItem.objects.create(archive=archive)
         item.values.set([civ])
+
+
+@shared_task
+def add_values_to_archive_item(*, archive_item_pk, civ_pks):
+    archive_item = ArchiveItem.objects.get(pk=archive_item_pk)
+    archive_item.values.clear()
```

Comment:
Do you really want to clear here? If so, we must change the function name as this is not what I would expect it to do. Must this always be called with the pks that you want to keep? I think there would be less room for future programming mistakes if we separated this out into add and remove operations.

Reply:
What I really wanted to do here is use `.set()`, but unfortunately that did not trigger the `m2mchanged` signal.

Reply:
Ooh wasn't aware that didn't trigger, we should check the codebase for usages of that. Here I would still prefer the explicit remove/add operations, should mean less work with what we do in the signal handler(s) too.